R/lint.R

```diff
@@ -174,12 +187,10 @@ lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings =
 #'   )
 #' }
 #' @export
-lint_dir <- function(path = ".", relative_path = TRUE, ..., exclusions = list("renv", "packrat"),
-                     pattern = rex::rex(
-                       ".", one_of("Rr"),
-                       or("", "html", "md", "nw", "rst", "tex", "txt"),
-                       end
-                     ),
+lint_dir <- function(path = ".", ...,
+                     relative_path = TRUE,
```

Comment:
We need a safeguard here too.
`lint_dir("...", FALSE)` would trigger the warning regarding `cache` once per file in "..." instead of alerting the actual problem that `relative_path` is no longer positional.

Reply:
this was intentional:

> I'll write it for lint() and we can decide whether to proceed for the others.

Wanted to check with you first that the implementation for `lint()` looks good